### DISCUSSION TITLE
How to pass the current video timestamp to the inner overlay widget?

### DISCUSSION THREAD

**Discussion by saurabhchalke** [jhomlala/betterplayer#356] [Q&A] [answered]
Hi, I really appreciate this player.
I have a use case where I try to draw overlay animations on top of the video based on the current timestamp.
I wanted to know:
1) How to get the current timestamp for the frame on screen?
2) How to pass the timestamp to the inner overlay widget defined in the configuration or is there a way the inner widget can keep streaming the value of the current timestamp?

Thanks in advance!

**Comment by jhomlala** [comment, marked as answer]
Hi, thanks for using BP.
1. betterPlayerController.videoPlayerController.value.duration
2. Use InheritedWidget or pass BetterPlayerController instance to inner widget.

**Reply by saurabhchalke** [reply]
Thank you for the quick help here, will try this out. :)
